### ISSUE TITLE
Game branching on asset.party

### ISSUE THREAD

**Issue by trende2001** [Facepunch/sbox-issues#3042]
### For?

S&Box, Other

### What can't you do?

It would be great to have a main and dev branch of a game on asset.party.
Why? If you want to do an internal/external playtest of your game on your developer branch (on github for example), people will need to download your local game assets using your upload speed. If you have a lot of people joining, this comes down to a slow halt.

Developers along with you would also have to redownload all of the assets for your game if they want to check out what you implemented, s&box already assumes that you don't have the assets to that game and redownloads them, wasting time and space

### How would you like it to work?

Uploading an experimental version of your game to a developer branch you create for your game on asset.party, having passworded branches, so on and so fourth. Kind of like betas and alphas before you all mash it into production

### What have you tried?

N/A

### Additional context

It would be genuinely awesome to download the developer version of the game from a blazing fast download server

**Comment by trende2001** [comment]
if you don't have the +branch convar specified, it would default to the production (main) branch

**Comment by SplatzyXD** [comment]
Would really help out in game testing.

**Comment by trende2001** [comment]
yeah, i believe there could be an option to link the github repo of your game and it would pull all of the branches from there, though you could choose to exclude a few branches from being shown if you'd like

**Comment by matekdev** [comment]
I'm actively working on a major overhaul to our systems in My Summer Cottage. One of these includes changing the structure of our game resources. Due to this, I'm unable to test locally anymore since the second client will always download the asset party version of the game resources.

I can obviously re-upload to asset.party, however, this will break the live version of our build.

**Comment by MrBrax** [comment]
> It'd be nice to have branching for this use case.

in addition to this, invite-only play permissions would be useful for the entire game or branches

**Comment by matekdev** [comment]
This is still a massive pain point.

Local multiplayer testing pulls all resources from the live build. So you are unable to do any testing locally without uploading to the live copy.

Either this build branching needs to be fixed introduced or local multiplayer testing should pull from your local copy.

**Comment by garrynewman** [comment]
We have a plan for this, but branching isn't really the solution to testing locally. When testing locally those games should work regardless of being on asset party or not.
